torchtitan/components/optimizer.py

```diff
@@ -177,8 +177,47 @@ def zero_grad(self) -> None:
         pass
 
 
+class FTOptimizersContainer(OptimizersContainer):
+    def __init__(
+        self,
+        model_parts: List[nn.Module],
+        optimizer_kwargs: Dict[str, Any],
+        name: str,
+        ft_manager: Optional["ft.Manager"],
+    ) -> None:
+        import torchft as ft
+
+        super().__init__(model_parts, optimizer_kwargs, name)
+
+        # Force to initialize the optimizer state so that `optim.step()`
+        # won't be called by state_dict() and load_state_dict().
+        _ = {
+            k: v
+            for sd in map(get_optimizer_state_dict, model_parts, self.optimizers)
+            for k, v in sd.items()
+        }
+        self.optimizers = [ft.Optimizer(ft_manager, optim) for optim in self.optimizers]
```

Comment:
How many optimizers are there usually? How does OptimizersContainer work? Does it call .zero_grad and .step multiple times?

With multiple torchft OptimizerWrappers we'll end up calling `start_quorum` and `should_commit` once for each wrapper which doesn't seem intended.

Maybe better to override zero_grad and step in this container class?

Reply:
There is only one `torchft.OptimizerWrapper` because only one optimizer is visible to `torch.OptimizerWrapper` -- OptimizersContainer.

Reply:
Am I misunderstanding what this line does?

```
[ft.Optimizer(ft_manager, optim) for optim in self.optimizers]
```

This creates one wrapper per optimizer in self.optimizers? It's guaranteed to only have one optimizer? Can we add an assert for that?

Reply:
yup, you are right. I confuse myself about this CTOR. Let me double check this.

Reply:
Fixed this issue. Thanks for noticing the bug.